### PR TITLE
Problem: BEP-10 goals are vague.

### PR DESCRIPTION
**Pull request by ldmberman**
Solution: Be very explicit about BEP-10 goals. Additionaly, be more explicit about choosing Go as a language. Add some minor refinements.

### CODE REVIEW

File: 10/README.md [hunk 1]
```diff
@@ -4,29 +4,70 @@ Name: A Strangler Application Approach to Rewriting Some Code in Go
...
 
-BigchainDB 2.0 Alpha 1 is now the composition of three different system (MongoDB, Tendermint, and BigchainDB itself). This setup generates different problems: a) managing multiple services together adds overhead and complicates deployment; b) the current design makes some calls going from BigchainDB to Tendermint to BigchainDB again, penalizing performance; c) there is data duplication between LevelDB and MongoDB; d) we need to maintain [`py-abci`][pyabci], while the official implementation of the [ABCI][abci] interface is in Golang; e) the logic to validate transactions is spread across multiple files and it's difficult to manage, and a heavy refactoring is needed; f) the configuration is spread across BigchainDB and Tendermint.
+BigchainDB 2.0 Alpha 1 is now the composition of three different system (MongoDB, Tendermint, and BigchainDB itself).
+
+This setup generates different problems:
+
+1. managing 3 services together adds operational overhead and complicates distribution;
+
+    Blockchain clients are commonly independent binaries. This is true for Bitcoin, Etherium, and Monero.
```
Comment (ttmc) [inline 10/README.md]: Ethereum

File: 10/README.md [hunk 2]
```diff
@@ -4,29 +4,70 @@ Name: A Strangler Application Approach to Rewriting Some Code in Go
...
+
+2. the current design raises certain performance concerns;
+
+    The TCP communications between BigchainDB and Tendermint, the transaction validation implemented in Python, and Crypto-Conditions implemented in Python might significantly contribute to the performance degradation. The detailed benchmarks are on the way.
+
+3. there is data duplication between LevelDB and MongoDB;
+
+    By tightly integrating with Tendermint, the default database backend can be replaced.
```
Comment (ttmc) [inline 10/README.md]: I'm not sure tight integration is needed before the LevelDB database can be replaced by the MongoDB database. All one would have to do is make a custom version of Tendermint Core, i.e. by changing the code of Tendermint Core, which is written in Go. Learning Go (and the internals of Tendermint Core) seems like the main thing blocking us from replacing LevelDB with MongoDB. The BEP helps us start learning Go.

File: 10/README.md (hunk 1 above)
Comment (ttmc) [inline 10/README.md]: Ethereum

File: 10/README.md (hunk 2 above)
Comment (ldmberman) [reply 10/README.md]: > All one would have to do is make a custom version of Tendermint Core, i.e. by changing the code of Tendermint Core, which is written in Go

You actually described what I am referring to by "tight integration". The term is probably not very intuitive.. I will try to change this part.

**Comment by ldmberman** [comment]
In general, if we agree on what we have and do not plan to add much to this BEP, I can also make it "Draft" here.

**Comment by ttmc** [comment]
Please don't make it "Draft" until we've done some of the things specified in the BEP. To see how BEP-2 (COSS) defines "draft", see:

https://github.com/bigchaindb/BEPs/tree/master/2#draft-beps

File: 10/README.md (hunk 2 above)
Comment (ttmc) [reply 10/README.md]: I don't understand but it doesn't matter right now. We can think about the whole "replace LevelDB" idea/design in another BEP.

**Comment by ldmberman** [comment]
I understand how it works for "standard" proposal, not that much for "informational" ones.

File: 10/README.md [hunk 3]
```diff
@@ -4,29 +4,70 @@ Name: A Strangler Application Approach to Rewriting Some Code in Go
...
 ## Overview
-In this BEP, we evaluate a rewrite of some parts of BigchainDB in another language, specifically Golang. The benefits are multiple: Golang it is a fairly simple language, easy to learn and read; it is shown to be efficient and faster than Python and can compile to a binary file; it is used in several big projects, and it's the language of choice for major blockchain projects: this simplifies code reuse if we need it in the future, and allows us a tighter integration with Tendermint.
+In this BEP, we evaluate a rewrite of some parts of BigchainDB in Golang.
+
+The main reason to choose Golang is:
+
+* Tendermint is written in Golang. Using Go allows to achieve the most efficient integration with Tendermint. It includes efficient ABCI communications, interactions with the database, and straightforward compilation of BigchainDB and Tendermint into a single binary.
+
+
+Other notes about Go:
```
Comment (kansi) [inline 10/README.md]: I would like to understand the reason behind writing these notes. IMHO the document should only have reason for leaning towards Golang in comparison to python. So, I think these notes should be contrasted with python.

**Comment by ldmberman** [comment]
@kansi I refined the description of the motivation behind choosing the language.

**Pull request merged**